Tests/tests/test_MotionDetection.cpp

```diff
@@ -96,107 +95,212 @@ TEST_F(MotionDetectionEventTest, getMotionDetectors)
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"1\" }"), response));
+    EXPECT_EQ(response,  string("{\"success\":true}"));
 }
 
-TEST_F(MotionDetectionEventTest, armmotiondetected)
+TEST_F(MotionDetectionEventTest, armmotiondetectedInvalid)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_ArmMotionDetector(::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](MOTION_DETECTION_Mode_t mode, std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"1\" }"), response));
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string(""));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("arm"), _T("{\"mode\":\"0\" }"), response));
+    EXPECT_EQ(response,  string(""));
 }
 
-TEST_F(MotionDetectionEventTest, armErrmotiondetected)
+TEST_F(MotionDetectionEventTest, disarm)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_DisarmMotionDetector(::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index) {
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"10\" }"), response));
-        EXPECT_EQ(response,  string("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string("{\"success\":true}"));
 }
 
-
-TEST_F(MotionDetectionEventTest, disarm)
+TEST_F(MotionDetectionEventTest, disarmInvalid)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_DisarmMotionDetector(::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_XX\"}"), response));
+    EXPECT_EQ(response,  string(""));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_DisarmMotionDetector(::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("disarm"), _T("{}"), response));
```

Comment:
is it returning ERROR_GENERAL due to invalid parameters or due to MOTION_DETECTION_RESULT_INDEX_ERROR? This is not clear from the test or its name. If a test verifies 2 different behaviors it's better be split into 2 tests